hyprbars/barDeco.cpp

```diff
@@ -540,11 +541,36 @@ PHLWINDOW CHyprBar::getOwner() {
     return m_pWindow.lock();
 }
 
-void CHyprBar::setHidden(bool hidden) {
-    if (m_bHidden == hidden)
-        return;
+void CHyprBar::updateRules() {
+    const auto PWINDOW = m_pWindow.lock();
+    auto rules = PWINDOW->m_vMatchedRules;
+    auto prev_m_bHidden = m_bHidden;
+    auto prev_m_bForcedTitleColor = m_bForcedTitleColor;
+
+    m_bForcedBarColor = std::nullopt;
+    m_bForcedTitleColor = std::nullopt;
+    m_bHidden = false;
+
+    for(auto& r : rules) {
+        applyRule(r);
+    }
 
-    m_bHidden = hidden;
+    if (prev_m_bHidden != m_bHidden) {
+        g_pDecorationPositioner->repositionDeco(this);
+    }
+    if (prev_m_bForcedTitleColor != m_bForcedTitleColor) {
+        m_bTitleColorChanged = true;
+    }
+}
 
-    g_pDecorationPositioner->repositionDeco(this);
+void CHyprBar::applyRule(const SWindowRule& r) {
+    auto arg = r.szRule.substr(r.szRule.find_first_of(' ') + 1);
+
+    if (r.szRule == "plugin:hyprbars:nobar") {
```

Comment:
raa

Reply:
Sorry, what does that stand for again?

Reply:
Probably meant that he saw {} around short ifs again.